tasks/sep/metal-001-MCP-L1/SignFromJson.s.sol

```diff
@@ -0,0 +1,448 @@
+        require(address(portalToCheck.superchainConfig()).code.length != 0, "6401");
+        require(EIP1967Helper.getImplementation(address(portalToCheck.superchainConfig())).code.length != 0, "6402");
+
+        require(portalToCheck.paused() == SuperchainConfig(proxies.SuperchainConfig).paused(), "6500");
+
+        require(portalToCheck.l2Sender() == Constants.DEFAULT_L2_SENDER, "6600");
+    }
+
+    /// @notice Asserts that the ProtocolVersions is setup correctly
+    function checkProtocolVersions() internal view {
+        console.log("Running assertions on the ProtocolVersions");
+
+        require(proxies.ProtocolVersions.code.length != 0, "6700");
+        require(EIP1967Helper.getImplementation(proxies.ProtocolVersions).code.length != 0, "6701");
+
+        ProtocolVersions protocolVersionsToCheck = ProtocolVersions(proxies.ProtocolVersions);
+        require(protocolVersionsToCheck.owner() == protocolVersionsOwner, "6800");
+        require(ProtocolVersion.unwrap(protocolVersionsToCheck.required()) == requiredProtocolVersion, "6900");
+        require(ProtocolVersion.unwrap(protocolVersionsToCheck.recommended()) == recommendedProtocolVersion, "7000");
+    }
+
+    /// @notice Asserts that the SuperchainConfig is setup correctly
+    function checkSuperchainConfig() internal view {
+        console.log("Running assertions on the SuperchainConfig");
+
+        require(proxies.SuperchainConfig.code.length != 0, "7100");
+        require(EIP1967Helper.getImplementation(proxies.SuperchainConfig).code.length != 0, "7101");
+
+        SuperchainConfig superchainConfigToCheck = SuperchainConfig(proxies.SuperchainConfig);
+        require(superchainConfigToCheck.guardian() == superchainConfigGuardian, "7200");
+        require(superchainConfigToCheck.guardian().code.length != 0, "7250");
+        require(superchainConfigToCheck.paused() == false, "7300");
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory /* simPayload */ )
+        internal
+        view
+        override
+    {
+        console.log("Running post-deploy assertions");
+
+        checkStateDiff(accesses);
+        checkSemvers();
+
+        checkSystemConfig();
+        checkL1CrossDomainMessenger();
+        checkL1StandardBridge();
+        checkL2OutputOracle();
+        checkOptimismMintableERC20Factory();
+        checkL1ERC721Bridge();
+        checkOptimismPortal();
+        checkProtocolVersions();
+        checkSuperchainConfig();
+
+        console.log("All assertions passed!");
+    }
+
+    // This method is not storage-layout-aware and therefore is not perfect. It may return erroneous
+    // results for cases like packed slots, and silently show that things are okay when they are not.
+    function isLikelyAddressThatShouldHaveCode(uint256 value) internal pure returns (bool) {
```

Comment:
We should turn these helper functions into a library

Reply:
@maurelian was looking into this. I think we could:
- Update the hardcoded known EOAs to read from the registry
- Update the checks against those EOAs to also include known account + slot numbers. The current approach would let you write a known EOA to *any* slot, which is too broad of a check. 

I don't think we should block this PR on that though